Content.Client/Mining/MiningOverlay.cs

```diff
@@ -0,0 +1,93 @@
+using System.Numerics;
+using Content.Shared.Mining.Components;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Shared.Enums;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
+
+namespace Content.Client.Mining;
+
+public sealed class MiningOverlay : Overlay
+{
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+    private readonly EntityLookupSystem _lookup;
+    private readonly SpriteSystem _sprite;
+    private readonly TransformSystem _xform;
+
+    private readonly EntityQuery<SpriteComponent> _spriteQuery;
+    private readonly EntityQuery<TransformComponent> _xformQuery;
+
+    public override OverlaySpace Space => OverlaySpace.WorldSpace;
+    public override bool RequestScreenTexture => true;
+
+    public MiningOverlay()
+    {
+        IoCManager.InjectDependencies(this);
+
+        _lookup = _entityManager.System<EntityLookupSystem>();
+        _sprite = _entityManager.System<SpriteSystem>();
+        _xform = _entityManager.System<TransformSystem>();
+
+        _spriteQuery = _entityManager.GetEntityQuery<SpriteComponent>();
+        _xformQuery = _entityManager.GetEntityQuery<TransformComponent>();
+    }
+
+    protected override void Draw(in OverlayDrawArgs args)
+    {
+        var handle = args.WorldHandle;
+
+        if (_player.LocalEntity is not { } localEntity ||
+            !_entityManager.TryGetComponent<MiningScannerViewerComponent>(localEntity, out var viewerComp) ||
+            !_xformQuery.TryComp(localEntity, out var localXform))
+            return;
+
+        if (viewerComp.LastPingLocation == null)
+            return;
+
+        var gridRot = localXform.GridUid == null ? 0 : _xformQuery.CompOrNull(localXform.GridUid.Value)?.LocalRotation ?? 0;
+        var rotationMatrix = Matrix3Helpers.CreateRotation(gridRot);
+        var scaleMatrix = Matrix3Helpers.CreateScale(new Vector2(1, 1));
```

Comment:
Did you check this worked zoomed in.

Also the viewport control has scale somewhere on it.